MetaMask/metamask-extension
# Picker network updates

## **Description**
In the [new designs for Amon Hen](https://github.com/MetaMask/metamask-extension/pull/21220/files#diff-afc7617dc1347d665d4f7a058e9ea8ac686f5ad520874a293f6e27e7c50f9568) the network picker(`PickerNetwork`) has been moved to inside of the tabs and needs to be full width. The current `PickerNetwork` requires some style updates to allow for this change

This PR updates the `PickerNetwork` component styles and props to allow for this updated design. It uses the pattern of inversion of control to allow for the label wrapper component to be accessed through a prop called `labelProps`. It also updates the unit tests to include coverage for this change as well as updates the documentation and stories to provide examples.

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Search `PickerNetwork` 
3. See that there are no visual regressions
4. Check the updated documentation and Width story

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/7ca4f2ea-63fb-4e12-a3ab-34e8890bbbfc

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/22036bca-7ebd-418d-8fae-c15f055bab0c

Other instances of `PickerNetwork` without visual regressions

<img width="327" alt="Screenshot 2023-10-10 at 10 29 55 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/8c43357b-6ddc-420c-b58f-1f1bc70ae40e"><img width="374" alt="Screenshot 2023-10-10 at 7 54 53 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/647e03bf-a775-4f03-bd70-787047876a04">

Showing new-network-info modal with some deprecated components replaced and flex box added

<img width="1512" alt="Screenshot 2023-10-11 at 3 12 43 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/2babc73c-f7c9-4df8-84b6-4e29dfd71872">

<img width="968" alt="Screenshot 2023-10-10 at 9 53 33 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/1dabfd5d-8aa2-4d65-b396-9a27eb22d9f9">
<img width="1512" alt="Screenshot 2023-10-10 at 10 29 24 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/34544b1f-bcc3-4668-a936-a4c151ca231a">


Test coverage 100%
<img width="1512" alt="Screenshot 2023-10-10 at 6 41 48 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/d85f2221-4d83-4c13-bd0a-d9e771bc0c76">


## **Related issues**
https://github.com/MetaMask/metamask-extension/pull/21220/files#diff-afc7617dc1347d665d4f7a058e9ea8ac686f5ad520874a293f6e27e7c50f9568

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
